### ISSUE TITLE
Amazon Appstore

### ISSUE THREAD

**Issue by DJaeger** [JusticeRage/ApkTrack#55]
I wanted to add Amazon Appstore as source, but recognized that it as 14 independend stores for different regions.

I thought of adding only the us one or all or automaticly the correct one for the user.

Adding all will make the souce chooser much longer, adding only one (US) store could have problems with apps only applied to one region (I don't know if Amazon do something like this) and automaticly determine the correct TLD could not be made with an static sources file.

What do you think is the best way?

**Comment by JusticeRage** [comment]
I think there are two ways to go: assume that the US Amazon app store will be good enough for everyone, or implement a sort of home-made localization capability for assets as described in this [StackOverflow question](https://stackoverflow.com/questions/5427933/localization-of-assets-files).

**Comment by DJaeger** [comment]
I searched a little bit.
By default your app is distributed worldwide, so in every amazon store.
But you can choose, if you want it to be only available in a specific store.
So we shouldn't only rely on the US store, as so much apps as possible should be found. At some time, earlier or later, someone will open an issue for an app, that is not found in the US store.

**Comment by DJaeger** [comment]
And remind, that the domain differs for countries, not for languages

**Comment by JusticeRage** [comment]
Good point about the difference between languages and countries. I believe that the aforementioned SO solution can be adapted to work with region instead of device language.

**Comment by DJaeger** [comment]
I think so too.

And I think the last solution is the best one for this app, as it don't need duplicate source files and is more open for other possible variables in the source file.
The app does in anyway processing on the file, which is the only disadventage of this solution.

**Comment by DJaeger** [comment]
Because it would be to simple:
http://www.amazon.com/gp/help/customer/display.html?nodeId=201357480#GUID-08425D2E-1083-4064-88A1-B2A4B13B43F2__SECTION_AA93D58D263A42B3BB428A4485FCE2B8

**Comment by JusticeRage** [comment]
Wow, this is just... wow. It won't be easy.

**Comment by DJaeger** [comment]
I think the best would be an option screen for changing repos.
That should allow to reorder, remove and add from predefined or custom.
This would allow to adding the stores you need (e.g. the correct amazon app store) and not more, which could safe time and bandwidht for searching updates, makes the list for choosing the source per app smaller and allows developers or tester to add sources for prereleases (like you do it too).
I think of a screen like the one for configuring DashClock Widget, with the addition that in the list of sources an additional button should be to add custom source.

**Comment by jawz101** [comment]
What about the XDA App Store or Aptoide repos?

**Comment by JusticeRage** [comment]
Could you provide a link to the XDA App Store?
I've just checked out Aptoide, but it seems like the URLs are hard to predict, so I probably won't be able to use it to provide direct links to APKs or get version information... However, it could probably be used as a search engine to locate APKs.

**Comment by jawz101** [comment]
I'm trying to figure it out.  They've closed source their app but say they're going to make it avail. on Github soon... though that's pretty uninformative :/
http://forum.xda-developers.com/android/apps-games/labs-t3241866
